src/components/forms/FormWorkflow.tsx

```diff
@@ -0,0 +1,150 @@
+import React from "react";
+import type { SyntheticEvent, Dispatch } from "react";
+import { Button, Form, Stepper, useToggle } from "@edx/paragon";
+
+import ConfigError from "../settings/ConfigError";
+// @ts-ignore
+import { useFormContext } from "./FormContext.tsx";
+import type {
+  FormFields,
+  FormFieldValidation,
+  FormContext,
+} from "./FormContext";
+// @ts-ignore
+import { setStepAction } from "./data/actions.ts";
+import { SUBMIT_TOAST_MESSAGE } from "../settings/data/constants";
+import { FormActionArguments } from "./data/actions";
+// @ts-ignore
+import UnsavedChangesModal from "../settings/SettingsLMSTab/UnsavedChangesModal.tsx";
+
+export type FormWorkflowButtonConfig<FormData> = {
+  buttonText: string;
+  onClick: (formFields: FormData) => Promise<void>;
+};
+
+type DynamicComponent = React.FunctionComponent | React.ComponentClass;
+
+export type FormWorkflowStep<FormData> = {
+  index: number;
+  stepName: string;
+  formComponent: DynamicComponent;
+  validations: FormFieldValidation[];
+  saveChanges: (FormData) => Promise<void>;
+  nextButtonConfig: FormWorkflowButtonConfig<FormData>;
+};
+
+export type FormWorkflowConfig<FormData> = {
+  steps: FormWorkflowStep<FormData>[];
+  getCurrentStep: () => FormWorkflowStep<FormData>;
+};
+
+export type FormWorkflowProps<FormData> = {
+  formWorkflowConfig: FormWorkflowConfig<FormData>;
+  onClickOut: (edited: boolean, msg?: string) => null;
+  formData: FormData;
+  dispatch: Dispatch<FormActionArguments>;
+};
+
+// Modal container for multi-step forms
+function FormWorkflow<FormData>({
+  formWorkflowConfig,
+  onClickOut,
+  dispatch,
+}: FormWorkflowProps<FormData>) {
+  const {
+    formFields,
+    currentStep: step,
+    hasErrors,
+    isEdited,
+  }: FormContext = useFormContext();
+  const [errorIsOpen, openError, closeError] = useToggle(false);
+  const [savedChangesModalIsOpen, openModal, closeModal] = useToggle(false);
+
+  const onCancel = () => {
+    if (isEdited) {
+      openModal();
+    } else {
+      onClickOut(false);
+    }
+  };
+
+  const onNext = async (event: SyntheticEvent) => {
+    await step?.nextButtonConfig.onClick(formFields as FormFields);
+    const nextStep: number = step.index + 1;
+    if (nextStep < formWorkflowConfig.steps.length) {
+      dispatch(setStepAction({ step: formWorkflowConfig.steps[nextStep] }));
+    } else {
+      // TODO: Fix
+      onClickOut(true, SUBMIT_TOAST_MESSAGE);
+    }
+  };
+
+  const stepBody = (step: FormWorkflowStep<FormData>) => {
+    const FormComponent: DynamicComponent = step?.formComponent;
+    return (
+      <Stepper.Step eventKey={step.index.toString()} title={step.stepName}>
+        <span>
+          {step && step?.formComponent && (
+            <>
```

Comment:
Why is the empty container needed around this? And also the span above it?

Reply:
It isn't needed, I'm pretty sure these were leftovers when I was debugging that portion of code.